lib/puppet-languageserver/message_handler.rb

```diff
@@ -45,6 +45,11 @@ def request_puppet_getversion(_, _json_rpc_message)
       )
     end
 
+    def request_puppet_getfacts(_, _json_rpc_message)
+      results = PuppetLanguageServer::PuppetHelper.get_all_facts(documents.store_root_path)
```

Comment:
This certainly seems sub-optimal, calling out to facter every time, but it's fine for now. We can refactor it later to use the object cache.